src/build-dep-graph.js

```diff
@@ -77,41 +75,14 @@ async function buildDepGraph({
   workspaceCwd,
   shouldPruneDeps = true,
 }) {
-  let workspacePackageJson = await readJson(path.join(workspaceCwd, 'package.json'));
-
-  let { workspaces } = workspacePackageJson;
-
-  let packagesGlobs;
-
-  let _1dFilesArray;
-  if (!workspaces) {
-    _1dFilesArray = (await execa('pnpm', ['recursive', 'exec', '--', 'node', '-e', 'console.log(process.cwd())'], { cwd: workspaceCwd })).stdout
-      .split(/\r?\n/)
-      .map(workspace => path.relative(workspaceCwd, workspace));
-
-    packagesGlobs = [];
-  } else {
-    packagesGlobs = workspaces.packages || workspaces;
-
-    let _2dFilesArray = await Promise.all(packagesGlobs.map(packagesGlob => {
-      return glob(packagesGlob, {
-        cwd: workspaceCwd,
-      });
-    }));
-
-    _1dFilesArray = Array.prototype.concat.apply([], _2dFilesArray);
-  }
-
-  let uniqueFiles = [...new Set(_1dFilesArray)];
-
-  let packageDirs = uniqueFiles.map(file => path.join(workspaceCwd, file));
+  let { packageDirs, packagesGlobs, rootWorkspacePackageJson } = await getWorkspacesPaths({ workspaceCwd });
 
   let workspaceMeta = {
     cwd: workspaceCwd,
     packagesGlobs,
   };
 
-  await firstPass(workspaceMeta, workspacePackageJson, packageDirs);
+  await firstPass(packageDirs, rootWorkspacePackageJson, workspaceMeta);
```

Comment:
Did the arg order need to change?

Reply:
nope, just alphabetical order

Reply:
I will revert it to keep changes minimal

Reply:
alphabetical order is bad IMO because if we add a forth than the order gets screwed up again? Can we put it back to its original order and also keep workspacePackageJson so the changeset gets smaller?